robottelo/ui/activationkey.py

```diff
@@ -113,107 +93,76 @@ def search_key_subscriptions(self, ak_name, subscription_name):
+                    )
             if content_view:
                 if env:
-                    strategy = locators["ak.env"][0]
-                    value = locators["ak.env"][1]
-                    element = self.wait_until_element((strategy, value % env))
-                    if element:
-                        element.click()
-                        self.wait_for_ajax()
-                    else:
-                        raise UIError(
-                            'Couldn not find the given env "{0}"'.format(env)
-                        )
+                    strategy, value = locators['ak.env']
+                    self.click((strategy, value % env))
                 # We need to select the CV, if we update the env and in this,
                 # case edit button disappears, but when we update only CV, then
                 # edit button appears; Following 'If' is just solving this
                 # purpose and hence no else required here
-                if self.wait_until_element(locators["ak.edit_content_view"]):
-                    self.find_element(locators["ak.edit_content_view"]).click()
-                    self.wait_for_ajax()
-                Select(self.find_element
-                       (locators["ak.edit_content_view_select"]
-                        )).select_by_visible_text(content_view)
-                self.find_element(locators["ak.save_cv"]).click()
-                self.wait_for_ajax()
+                if self.wait_until_element(locators['ak.edit_content_view']):
+                    self.click(locators['ak.edit_content_view'])
+                Select(
+                    self.find_element(locators['ak.edit_content_view_select'])
+                ).select_by_visible_text(content_view)
+                self.click(locators['ak.save_cv'])
         else:
             raise UIError(
                 'Could not update the activation key "{0}"'.format(name)
             )
 
     def delete(self, name, really):
-        """
-        Deletes an existing activation key
-        """
-
+        """Deletes an existing activation key."""
         element = self.search_key(name)
 
         if element:
             element.click()
             self.wait_for_ajax()
-            self.wait_until_element(locators["ak.remove"]).click()
-            self.wait_for_ajax()
+            self.click(locators['ak.remove'])
             if really:
-                self.wait_until_element(common_locators["confirm_remove"]
-                                        ).click()
+                self.click(common_locators['confirm_remove'])
             else:
-                self.wait_until_element(locators["ak.cancel"]).click()
+                self.click(locators['ak.cancel'])
 
     def associate_product(self, name, products):
-        """
-        associate an existing product with activation key
-        """
-
+        """Associate an existing product with activation key."""
         element = self.search_key(name)
 
         if element:
```

Comment:
Will change, thanks for pointing this out. I was focusing on change the click lines and not worried about this kind of refactor.